.pre-commit-config.yaml

```diff
@@ -0,0 +1,5 @@
+repos:
```

Comment:
I think it is unnecessary.  We don't need pre-commit hooks.

Reply:
The `pre-commit` hooks are not mandatory for anyone; placing the file here is simply a convenience for anyone that _does_ use `pre-commit`, and this is because it's impossible to force someone to put the `.git/hooks/pre-commit` script into their repository.

Is the fear here that placing this file here tacitly supports the pre-commit project?

Reply:
Yes. If anyone searches for a specific use case of this file, they will find `pre-commit`.

But for this tool, it is not really needed by the ingress-nginx project. This may confuse some people.

Reply:
I don't really agree with this because, again, the `.pre-commit-config.yaml` by being in the repository isn't forcing anyone to use that tool; it's simply there as a convenience for people who do.

But I've removed it 🤷

Reply:
Thanks! This is one of my concerns.